File: tests/code/corpus/langs/perl/registry.pl
package Geometry::Registry;

use strict;
use Geometry::Shapes;

sub new {
    my ($class) = @_;
    return bless { items => [] }, $class;
}

sub add {
    my ($self, $item) = @_;
    push @{ $self->{items} }, $item;
}

sub seed {
    my ($self) = @_;
    $self->add(1);
}

1;

File: tests/code/corpus/langs/java/Registry.java
package geometry;

import java.util.ArrayList;

class Registry {
    private final ArrayList<Object> items = new ArrayList<>();

    Registry() {}

    void add(Object item) {
        items.add(item);
    }

    void seed() {
        add(Factory.makeCircle(1.0));
    }
}
